### PR TITLE
Allow to get the team icon for open teams

### PR DESCRIPTION
**Pull request by enahum**

### CODE REVIEW

File: api4/team.go [hunk 1]
```diff
@@ -741,15 +741,16 @@ func getTeamIcon(c *Context, w http.ResponseWriter, r *http.Request) {
...
-		c.SetPermissionError(model.PERMISSION_VIEW_TEAM)
-		return
-	}
-
 	if team, err := c.App.GetTeam(c.Params.TeamId); err != nil {
 		c.Err = err
 		return
 	} else {
+		if !c.App.SessionHasPermissionToTeam(c.Session, c.Params.TeamId, model.PERMISSION_VIEW_TEAM) &&
+			team.Type != model.TEAM_OPEN {
```
Comment (jwilander) [inline api4/team.go]: Because of the screwiness with the team model I think this needs to be `(team.Type != model.TEAM_OPEN || team.AllowOpenInvite)`

**Comment by grundleborg** [comment]
This is a bit ugly with advanced permissions, but I think it's fine for now. We'll make a proper fix later (probably by distinguishing between "VIEW_OPEN_TEAMS" and "VIEW_CLOSED_TEAMS" or something).

**Comment by enahum** [comment]
@jwilander done

**Comment by jwilander** [comment]
@enahum can you also update this line https://github.com/mattermost/mattermost-api-reference/blob/master/v4/source/teams.yaml#L683 ?

**Pull request merged**